src/aal/aal.h

```diff
@@ -46,7 +46,7 @@ namespace snmalloc
      */
     static inline uint64_t tick()
     {
-#if __has_builtin(__builtin_readcyclecounter) && \
+#if __has_builtin(__builtin_readcyclecounter) && !defined(__CHERI__) && \
```

Comment:
Reading the later code, is it that `__builtin_readcyclecounter` is not sufficiently serialising on MIPS?  If so, we probably want to refactor the callers of this, rather than have an  `#ifdef` here.

On a related note, please avoid using `__CHERI__` for things that are MIPS specific, we want to be able to use `__CHERI__` for CHERI/RISC-V and Morello.

Reply:
This may be stale; I'll check if `__builtin_readcyclecounter` does the right thing for us.